### DISCUSSION TITLE
Versions of the Dependencies

### DISCUSSION THREAD

**Discussion by rajavishah** [bilayer-containers/bilayers#14] [Q&A] [answered]
Many of our dependencies currently specified in the Dockerfiles have newer versions available. Should we consider updating to the latest versions of the following dependencies in our Dockerfiles?

- [ ] python (docker image)- should we use the latest one?
- [ ] scipy
- [ ] scikit-image
- [ ] gradio
- [ ] gradio_client
- [ ] huggingface_hub
- [ ] pydantic

@gnodar01 We can change all these later too, as of now just dumping the questions popped in my mind

**Comment by gnodar01** [comment, marked as answer]
For python, most or perhaps even all of the algos we wrap will be python-based. That means we should not install our own python interpreter, but rely on the one installed by the wrapee.

For scipy, scikit-image, that's just our sample threshold algo. We're just using that as an MVP, and in the end we won't be publishing it to end users, since its not really useful. We can just keep those versions.

For gradio, gradio_client, huggingface_hub and pydantic, we probably want to use the latest versions (since that will change over time). However we don't want to use the `latest` tag. Just figure out the most recent versions of each, and hard pin it to that version in the dockerfile.

In general, for our dockerfiles, we want hard pins.

**Reply by rajavishah** [reply]
Yes, moved the described one to their latest versions.
